### PR TITLE
right-click player callback

### PR DESCRIPTION
**Pull request by runsy**
[https://forum.minetest.net/viewtopic.php?f=7&t=20722](https://forum.minetest.net/viewtopic.php?f=7&t=20722)

### CODE REVIEW

File: src/content_sao.cpp [hunk 1]
```diff
@@ -1325,6 +1325,15 @@ u16 PlayerSAO::punch(v3f dir,
 	return hitparams.wear;
 }
 
+void PlayerSAO::rightClick(ServerActiveObject *clicker)
+{
+	PlayerSAO *playersao = m_player->getPlayerSAO();
+
+	if (isAttached())
```
Comment (SmallJoker) [inline src/content_sao.cpp]: Should also be allowed for attached players. You can punch them, so why not right-click?

**Comment by paramat** [comment]
This will need documentation in lua_api.txt, although that can be done later when the PR is in its final form and close to a merge.

File: builtin/game/register.lua [hunk 2]
```diff
@@ -612,6 +612,7 @@ core.registered_on_modchannel_message, core.register_on_modchannel_message = mak
 core.registered_on_auth_fail, core.register_on_auth_fail = make_registration()
 core.registered_on_player_inventory_actions, core.register_on_player_inventory_action = make_registration()
 core.registered_allow_player_inventory_actions, core.register_allow_player_inventory_action = make_registration()
+core.registered_on_rightclickplayers, core.register_on_rightclickplayer = make_registration()
```
Comment (nerzhul) [inline builtin/game/register.lua]: i would like to see the same callback on CSM :)

**Comment by SmallJoker** [comment]
No action from author for several months. Possibly replaced by #9569.

**Pull request closed without merging**